tests/test_integration.py

```diff
@@ -567,3 +567,95 @@ def test_entry_point_targeting():
 
     stdout, rc = run_simple_pex(pex_out_path)
     assert 'usage: autopep8'.encode() in stdout
+
+
+def test_interpreter_selection_using_os_environ_for_bootstrap_reexec():
+  """
+  This test is a special test meant to verify proper function of the
+  pex bootstrapper's interpreter selection logic and validate the corresponding
+  bugfix. More details on the nature of the bug can be found at:
+  https://github.com/pantsbuild/pex/pull/441
+  """
+  with temporary_dir() as td:
+    pexrc_path = os.path.join(td, '.pexrc')
+
+    # Select pexrc interpreter versions based on test environemnt.
+    # The parent interpreter is the interpreter we expect the parent pex to 
+    # execute with. The child interpreter is the interpreter we expect the 
+    # child pex to execute with. 
+    if (sys.version_info[0], sys.version_info[1]) == (3, 6):
+      parent_interpreter_version = '3.6.2'
+    else:
+      parent_interpreter_version = '2.7.10'
+    if (sys.version_info[0], sys.version_info[1]) == (3, 6):
+      child_interpreter_version = '3.6.3'
```

Comment:
could probably combine these into one block like:

```
if (sys.version_info[0], sys.version_info[1]) == (3, 6):
  parent_interp...
  child_interp...
else:
  parent..
  child...
```

since they both use the same if clause.